src/layers/singleLayer.jl

```diff
@@ -62,62 +63,65 @@ function initTheta(this::singleLayer{T}) where {T <: Number}
 end
 
 
-function Jthetamv(this::singleLayer{T},dtheta::Array{T},theta::Array{T},Y::Array{T},tmp) where {T <: Number}
-    dA             = tmp[2]
-    nex            = div(length(Y),nFeatIn(this))
-    Y              = reshape(Y,:,nex)
+function Jthetamv(this::singleLayer{T},dtheta::Array{T},theta::Array{T},Yin::Array{T},tmp) where {T <: Number}
+    dA::Array{T,2}             = tmp[2]
+    nex            = div(length(Yin),nFeatIn(this))
+    Y              = reshape(Yin,:,nex)
 
     th1,th2,th3,th4     = splitWeights(this,theta)
     dth1,dth2,dth3,dth4 = splitWeights(this,dtheta)
-    dZ = Jthetamv(this.K,dth1,th1,Y) .+ this.Bin*dth2
+    dZ::Array{T,2} = Jthetamv(this.K,dth1,th1,Y) .+ this.Bin*dth2
```

Comment:
I would assume this means `Jthetamv(::ConvGEMMKernel`) is unstable? Or `this.Bin*dth2`. Either one, we should follow up